src/components/icon-extention-with-title/IconExtentionWithTitle.tsx

```diff
@@ -1,38 +1,93 @@
-import { FC } from 'react'
+import { FC, useState, useRef, useEffect, ChangeEvent } from 'react'
 import { useTranslation } from 'react-i18next'
-
+import IconButton from '@mui/material/IconButton'
+import ClearIcon from '@mui/icons-material/Clear'
+import DoneIcon from '@mui/icons-material/Done'
+import InputBase from '@mui/material/InputBase'
 import Box from '@mui/material/Box'
 
 import TitleWithDescription from '~/components/title-with-description/TitleWithDescription'
-import { styles } from '~/components/icon-extention-with-title/IconExtentionWithTitle.styles'
 
+import { snackbarVariants } from '~/constants'
+import { styles } from '~/components/icon-extention-with-title/IconExtentionWithTitle.styles'
 import { convertBytesToProperFormat } from '~/utils/helper-functions'
 
 interface IconExtentionWithTitleProps {
   title: string
   description?: number
+  isEditable?: boolean
+  onCancel?: () => void
+  onSave?: (fileName: string) => Promise<void>
 }
 
 const IconExtentionWithTitle: FC<IconExtentionWithTitleProps> = ({
   title,
-  description
+  description,
+  isEditable = false,
+  onCancel,
+  onSave
 }) => {
   const { t } = useTranslation()
+  const inputRef = useRef<HTMLInputElement | null>(null)
+  const initialInputValue = title.substring(0, title.lastIndexOf('.'))
+  const [inputValue, setInputValue] = useState<string>(initialInputValue)
+  const [fileExtension] = title.split('.').reverse()
 
-  const fileExtension = title.slice(title.lastIndexOf('.') + 1)
   const convertDescription = (descriptionNumber: number) => {
     const { size, unit } = convertBytesToProperFormat(descriptionNumber)
     return size + ' ' + t(`common.${unit}`)
   }
 
+  const onChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setInputValue(e.target.value)
+  }
+
+  const onCancelHandler = () => {
+    onCancel?.()
+  }
+
+  const onSaveHandler = () => {
+    const trimmedValue = inputValue.trim()
+    const value = trimmedValue !== initialInputValue ? trimmedValue : ''
+    onSave && void onSave(value)
```

Comment:
I guess, that you can leave just 
`const value = trimmedValue ? inputValue.trim() : ''`

Reply:
I check if trimmedValue is equal to initial, so, if user didn`t change the name, I set it to an empty string, and if value is an empty string the new request will not be called

Reply:
Perhaps it is unnecessary to permit saving a file without a name.

Reply:
It will not save the file without a name or without changing, onSave function check inside if the value is empty, if not, it will make a request